alpakka/src/main/scala/org/scanamo/ops/retrypolicy/RetryPolicy.scala

```diff
@@ -39,39 +25,30 @@ sealed abstract class RetryPolicy extends Product with Serializable { self =>
   }
 
   final def update: RetryPolicy = self match {
-    case policy @ Constant(_, numberOfRetries)  => policy.copy(numberOfRetries = numberOfRetries - 1)(policy.scheduler)
-    case policy @ Linear(_, numberOfRetries, _) => policy.copy(numberOfRetries = numberOfRetries - 1)(policy.scheduler)
-    case policy @ Exponential(_, numberOfRetries, _) =>
-      policy.copy(numberOfRetries = numberOfRetries - 1)(policy.scheduler)
+    case policy: Constant            => policy
+    case policy: Linear              => policy
+    case policy: Exponential         => policy
     case Max(retries)                => Max(retries - 1)
     case And(thisPolicy, thatPolicy) => And(thisPolicy.update, thatPolicy.update)
     case Or(thisPolicy, thatPolicy)  => Or(thisPolicy.update, thatPolicy.update)
     case retryPolicy                 => retryPolicy
   }
 
+  final val retries: Int = self match {
+    case Max(retries) => retries
+    case And(x, y)    => Math.min(x.retries, y.retries)
```

Comment:
This should be case And(x, y)    => Math.max(x.retries, y.retries), if we want to complete the retries for both

Reply:
Good catch!